### ISSUE TITLE
feat: spritemap is on cache?

### ISSUE THREAD

**Issue by matteobertoldo** [cascornelissen/svg-spritemap-webpack-plugin#134]
Hi! 👋 I use this plugin with the latest version of `webpack` (`>=5.0.0`)...it works very well!...
but every time I save a file other than a `.svg` extension a spritemap is generated again..it seems that the final bundle is not in cache. What happens with the `.scss` & `.js` files instead (the first two modules are cached as you can see from the screenshot)

<img width="511" alt="hero-cli-spritemap-cache" src="https://user-images.githubusercontent.com/15775323/100736991-db94a500-33d3-11eb-9525-2a99cbc96d39.png">

Here is my configuration:

```js
new SVGSpritemapPlugin('src/svg/**/*.svg', {
  output: {
     filename: 'svg/spritemap.svg'
  },
  sprite: {
    prefix: 'icon-',
    generate: {
      title: false
    }
  }
});
```

I forgot any settings? Thanks 😊

**Comment by cascornelissen** [comment]
Hmm, to be honest I have actually never looked into this. I was kind of expecting Webpack to handle this internally but we might need to add some logic to make it clear to the compiler that nothing changed somehow.

If you (or anyone else reading this) want to; feel free to dig into this and submit a PR. If nothing happens with it I might check it out once I have some free time again.

**Comment by cascornelissen** [comment]
@matteobertoldo, is there any specific setup you use to get these `[cached]` indicators in your output? I've been trying to get them to show up for some simple JS files but they're never in the stats output 😅

**Comment by matteobertoldo** [comment]
@cascornelissen here is my [configuration](https://github.com/matteobertoldo/freedomhood/blob/master/webpack.config.js)! 😊

Now I try to set the `cache` settings to `true` and update you.

**Comment by cascornelissen** [comment]
@matteobertoldo, I'm pretty sure I know where the problem lies but it boils down to the fact that an asset (the spritemap SVG) can't have cache dependencies within the Webpack internals. Which is strange, because the stats printing logic does check whether the `cached` property on the asset object is truthy. I'm digging some deeper into this 🕵🏼 

Do your projects have any other **assets** (not modules) that do get cached properly?

**Comment by matteobertoldo** [comment]
unfortunately I only have `.scss`, `.js` and `.svg` files to compile. I have no other types of _assets_ that come from other plugins, so I can keep track of whether it manages to keep the cache or not...

**Comment by cascornelissen** [comment]
Phew, that was quite the rabbit hole but the changes in 4db4ca6 should improve this and it also includes some other minor performance improvements. Since this is quite a large refactor please test it out thoroughly in your setup @matteobertoldo (also with Webpack 4 if that's feasible for you).

You can install these recent changes directly from GitHub using the following command.

```
npm install cascornelissen/svg-spritemap-webpack-plugin#master
```

Let me know if it solves the issue on your side ✌🏼

**Comment by matteobertoldo** [comment]
@cascornelissen mmmh: `HookWebpackError: compilation.options.plugins.filter(...).flatMap is not a function`

<img width="956" alt="Schermata 2020-12-04 alle 14 50 26" src="https://user-images.githubusercontent.com/15775323/101171989-d7b98a80-3640-11eb-8620-6a783dbcdbd3.png">

<img width="964" alt="Schermata 2020-12-04 alle 14 50 38" src="https://user-images.githubusercontent.com/15775323/101171994-d9834e00-3640-11eb-9e13-172a5cf47d07.png">

**Comment by cascornelissen** [comment]
Whoops, that's my bad, you're probably on an older version of Node.js. Commit 1ebaa74 fixes this, could try again? 😅

**Comment by matteobertoldo** [comment]
@cascornelissen well done! 🙇🏽‍♂️💯 Perfect! waiting for the next release! 😊

Thank u so much!

<img width="457" alt="cli" src="https://user-images.githubusercontent.com/15775323/101363973-506b5180-38a2-11eb-8991-51fcdbbbc08e.png">

**Comment by cascornelissen** [comment]
Good to hear! I'm aiming to combine this together with the fix for #135 in the next release. I'll post a comment in both issues when the release is available ✌🏼

**Comment by cascornelissen** [comment]
These changes were included in [`3.8.0`](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/releases/tag/v3.8.0), thanks once more for reporting the issue! 🚀
